### PR TITLE
Add gradient accumulation to CrossEncoder

### PR DESCRIPTION
**Pull request by hugoabonizio**
This PR adds gradient accumulation to CrossEncoder training, which might be very useful when using larger models or training with larger batch sizes.

### CODE REVIEW

**Comment by tomaarsen** [comment]
Apologies for the delay. This behaviour has been added in the Sentence Transformers v4.0 release, which introduced the CrossEncoderTrainer.

- Tom Aarsen
